cstar/roms/runtime_settings.py

```diff
@@ -0,0 +1,806 @@
+    elif abs(val) < 1e-2 or abs(val) >= 1e4:
+        return f"{val:.6E}".replace("E+00", "E0")
+    else:
+        return str(val)
+
+
+def _format_path(path: Path) -> str:
+    return str(path)
+
+
+def _format_other(other: str | int) -> str:
+    return str(other)
+
+
+################################################################################
+
+
+class ROMSRuntimeSettingsSection(BaseModel):
+    section_name: ClassVar[str]
+    multi_line: ClassVar[bool] = False
+    key_order: ClassVar[list[str]]
+
+    def __init__(self, *args, **kwargs):
+        if args:
+            super().__init__(**{k: args[i] for i, k in enumerate(self.key_order)})
+        else:
+            super().__init__(**kwargs)
+
+    @model_validator(mode="wrap")
+    @classmethod
+    def validate_from_lines(cls, data, handler: ModelWrapValidatorHandler):
+        # if the class gets a list of strings as it's init, assume it's coming in as a line
+        # from the roms.in file, and try to parse it as such. if that fails, or if it's not a list
+        # when it comes in, do the usual init process.
+
+        if isinstance(data, list) and all(isinstance(v, str) for v in data):
+            try:
+                return cls.from_lines(data)
+            except Exception as e:
+                log.debug(
+                    f"{cls.__name__}.validate_from_lines: failed to parse from lines: {data!r} – {e}"
+                )
+        return handler(data)
+
+    @property
+    def value_joiner(self):
+        return "\n    " if self.multi_line else "    "
+
+    @model_serializer
+    def default_serializer(self) -> str:
+        # Make a dictionary out of the attributes:
+        data = {k: getattr(self, k) for k in self.key_order}
+
+        section_values_as_list_of_str = []
+        # Check formatting of values
+        for value in data.values():
+            match value:
+                case list() if all(isinstance(v, float) for v in value):
+                    section_values_as_list_of_str.append(_format_list_of_floats(value))
+                case list() if all(isinstance(v, Path) for v in value):
+                    section_values_as_list_of_str.append(
+                        _format_list_of_paths(value, multi_line=self.multi_line)
+                    )
+                case list():
+                    section_values_as_list_of_str.append(_format_list_of_other(value))
+                case float():
+                    section_values_as_list_of_str.append(_format_float(value))
+                case Path():
+                    section_values_as_list_of_str.append(_format_path(value))
+                case None:
+                    continue
+                case _:
+                    section_values_as_list_of_str.append(_format_other(value))
+        section_values_as_single_str = self.value_joiner.join(
+            section_values_as_list_of_str
+        )
+
+        section_header = f"{self.section_name}: {' '.join(data.keys())}\n"
```

Comment:
any reason we're not using the `value_joiner` method to ensure consistent spacing?